docs/ApiGuidelines.md

```diff
@@ -0,0 +1,77 @@
+# API Guidelines
+
+## APIs
+
+There are three types of APIs: 
+
+1. Public
+2. Experimental
+3. Internal
+
+### Public APIs
+
+Public APIs are the supported APIs that apps can use. 
+
+Public API requirements: 
+
+* All public APIs must have at least one test case.
+* All public APIs must be in a header file that is published to a location under the SDK output directory.
+* All public APIs must be fully documented using doxygen markup in the public header files
+* All public APIs must be used in some sample code or snippet. Ideally, the sample
+  should be one that is compiled as part of the build, and uses no experimental APIs. For rarely 
+  used public APIs, the sample might simply be a snippet in the doxygen comments.
+* Public APIs cannot have source-breaking changes across Open Enclave versions. For platforms 
```

Comment:
Between major versions, I think it should be acceptable that we have public API source-breaking changes. It is not ideal, but is seen in many popular open source libraries.

Reply:
@dthaler do we want to specify "between Open Enclave minor versions"?

Reply:
never-mind. Deferring discussion to below.